File: src/models/Role.ts
import { Schema, model, Types, Document } from "mongoose";

export interface Role {
	id: bigint;
	guild_id: bigint;
	color: number;
	hoist: boolean;
	managed: boolean;
	mentionable: boolean;
	name: string;
	permissions: bigint;
	position: number;
	tags?: {
		bot_id?: bigint;
	};
}

export interface RoleDocument extends Document, Role {
	id: bigint;
}

export const RoleSchema = new Schema({
	id: Types.Long,
	guild_id: Types.Long,
	color: Number,
	hoist: Boolean,
	managed: Boolean,
	mentionable: Boolean,
	name: String,
	permissions: Types.Long,
	position: Number,
	tags: {
		bot_id: Types.Long,
	},
});

export const RoleModel = model<RoleDocument>("Role", RoleSchema, "roles");

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Role.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("roles")]
public partial class Role
{
    [Key]
    [Column("id", TypeName = "character varying")]
    public string Id { get; set; } = null!;

    [Column("guild_id", TypeName = "character varying")]
    public string GuildId { get; set; } = null!;

    [Column("color")]
    public int Color { get; set; }

    [Column("hoist")]
    public bool Hoist { get; set; }

    [Column("managed")]
    public bool Managed { get; set; }

    [Column("mentionable")]
    public bool Mentionable { get; set; }

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("permissions", TypeName = "character varying")]
    public string Permissions { get; set; } = null!;

    [Column("position")]
    public int Position { get; set; }

    [Column("icon", TypeName = "character varying")]
    public string? Icon { get; set; }

    [Column("unicode_emoji", TypeName = "character varying")]
    public string? UnicodeEmoji { get; set; }

    [Column("tags")]
    public string? Tags { get; set; }

    [Column("flags")]
    public int Flags { get; set; }

    [Column("colors")]
    public string Colors { get; set; } = null!;

    [ForeignKey("GuildId")]
    [InverseProperty("Roles")]
    public virtual Guild Guild { get; set; } = null!;

    [ForeignKey("RoleId")]
    [InverseProperty("Roles")]
    public virtual ICollection<Member> Indices { get; set; } = new List<Member>();

    [ForeignKey("RolesId")]
    [InverseProperty("Roles")]
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
}
